api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -186,6 +186,22 @@ type AzureManagedControlPlaneSpec struct {
 	// For authentication with Azure Container Registry.
 	// +optional
 	KubeletUserAssignedIdentity string `json:"kubeletUserAssignedIdentity,omitempty"`
+
+	// HTTPProxyConfig is the HTTP proxy configuration for the cluster.
+	// Immutable.
+	// +optional
+	HTTPProxyConfig *HTTPProxyConfig `json:"httpProxyConfig,omitempty"`
+}
+
+type HTTPProxyConfig struct {
```

Comment:
should any/all of these be marked as optional?

Reply:
Good point, however, since these are native types themselves (e.g. `*string`, `*[]string`) should we care to mark them explicitly optional/non-optional ?

Reply:
If it's non-optional, we don't need to do anything, but we do want to explicitly include the `// +optional` comment.

Reply:
Ah I see. Makes sense.